scripts/util/math.ts

```diff
@@ -1,40 +1,36 @@
-/**
- * Utility functions for Javascript.
- * Could be ported to C++ and exposed globally in the future.
- */
+export function magnitude(vec: vec2 | vec3): number {
+	return Math.sqrt(sumOfSquares(vec));
+}
 
-/**
- * 2D vector object with x and y member variables
- */
-type Vec2D = { x: number; y: number };
+export function magnitude2D(vec: vec2): number {
+	return Math.sqrt(sumOfSquares2D(vec));
+}
 
-/**
- * 3D vector object with x, y, and z member variables
- */
-type Vec3D = { x: number; y: number; z: number };
-type Vector = Vec2D | Vec3D;
+export function magnitude3D(vec: vec3): number {
+	return Math.sqrt(sumOfSquares3D(vec));
+}
 
-/**
- * 2D length of input vector
- */
-function getSize2D(vec: Vector): number {
-	return Math.sqrt(getSizeSquared2D(vec));
+// Note: Math.hypot performs additional bounds checking on V8 which which makes it considerably
+// slower than below implementations.
+export function sumOfSquares2D(vec: vec2): number {
```

Comment:
same here, `vec` should be allowed to be `vec3 | vec2`. In the generic case, the 3D magnitude of a `vec3` will be returned, but if `magnitude2D()` is called on a `vec3` explicitly, the XY-projected magnitude should be returned.